src/sections/create-dataset/CreateDatasetContext.tsx

```diff
@@ -0,0 +1,30 @@
+import { createContext, useContext } from 'react'
+import { CreateDatasetFormFields } from '../../dataset/domain/useCases/createDataset' // Importing the FormFields type
+
+// Define the context and its interface
+interface FormContextInterface {
+  formState: CreateDatasetFormFields
+  updateFormState: (newState: CreateDatasetFormFields) => void
+}
+
+// Define default values for the context
+const defaultFormState: CreateDatasetFormFields = {
+  // Initialize with default values for your form fields
+  createDatasetTitle: ''
+}
+
+const defaultContext: FormContextInterface = {
+  formState: defaultFormState,
+  updateFormState: () => {
+    // This is a no-op function since the default context shouldn't update anything
+  }
+}
```

Comment:
Place Submit and Validate logic here, pass props down to Form component to consume. Remove provider and context so that only hook remains.

Form states from component will be placed here.